### ISSUE TITLE
Add visualization for timelines (from 1178)

### ISSUE THREAD

**Issue by gigamorph** [project-lux/lux-frontend#15]
Problem Description: LUX has a requirement for visualizations.  The existing wireframes show this: https://nj46vq.axshare.com/#id=63klng&p=entity_-_person_group&g=1. We also need requirements and designs for visualizations.

Expected Behavior:
Render timelines using a visualization and improve the list view for timelines

Breaking out list view to:
[New ticket - 52](https://github.com/project-lux/lux-frontend/issues/52)
[Old ticket for reference - 1872](https://git.yale.edu/lux-its/lux-web/issues/1872) 

- [x] Visualization Requirements
- [x] Wireframes - Visualization
- [x] Mockup - Visualization
- [x] Question - is #52 required for this ticket to move forward? 
Answer: No

Links:
- https://www.bugherd.com/projects/284041/tasks/1708
- old tix: https://git.yale.edu/lux-its/lux-web/issues/1178

Wireframe (design comp below in thread)
![image](https://github.com/project-lux/lux-frontend/assets/160149818/4ebc94ef-4c8b-4676-ab9d-8a1b67dd5f0b)

Example of data from:
https://lux-front-tst.collections.yale.edu/view/person/c225d5e4-8767-4a25-802b-af054d5e8f52 
![image](https://github.com/project-lux/lux-frontend/assets/160149818/fed5588e-dfae-4b1a-b336-9c59dff4e1bf)

**Comment by kamerynB** [comment]
@prowns @jffcamp @ar2674 This issue requires better requirements. I think if the list view changes then it will impact the manner in which the frontend displays this data. The graph would appear the same but depending on the list view, the data may be in a different structure than what it is currently.

**Comment by ar2674** [comment]
@kamerynB are you saying we should have better requirements for what the list view should look like upon a user selecting a specific date within the timeline? 

Wanted to make sure I understand.

**Comment by roamye** [comment]
Bringing this to IT meeting for discussion.

**Comment by kamerynB** [comment]
Update 3/28/24:
The timeline graph can be worked on and is not blocked by the requirements for #52.

**Comment by kamerynB** [comment]
Just one question on this. I started working on a proof of concept but with a stacked bar chart. There are two wireframes and one has the option to view each relationship but the other shows all relationships. Are both functionalities required? I would think we'd only want one.

Here is the proof of concept with both dummy data, dummy labels, and random styling. This was my attempt to work with Recharts which I like so far.
<img width="906" alt="Screenshot 2024-04-30 at 4 18 32 PM" src="https://github.com/project-lux/lux-frontend/assets/140102078/88bfb32d-3f35-4b6e-b461-b551fa56cfbd">

**Comment by miketullo95** [comment]
@kamerynB I don't believe I have any info on this, this may have been a heather wireframe. I think generally its fine to have the info stacked, my only question would be accessibility.

**Comment by kamerynB** [comment]
Update 5/8:
The first iteration of the timeline has been deployed to DEV. Please feel free to leave feedback here.
@prowns @jffcamp @roamye

**Comment by roamye** [comment]
Feedback based on the time line visualization on [Andy Warhol ](https://lux-front-dev.collections.yale.edu/view/person/34f4eec7-7a03-49c8-b1be-976c2f6ba6ba)page:

- when you adjust the date to a year after 1945 and before 2023 BUT does not have a record for that specific year the page will render an error boundary. (Ex: 1960) Should we allow the visualization to go to the next year which has a record to prevent an error boundary? (1962) 
- what is zoom in supposed to do? It is not greyed out but when I click on it (expanding/not expanding the visualization) it doesn't zoom into the graph. 
- Should the key below the graph be interactive? When a user selects a bar, it will take them to a search result that has the most proportion and then the one with the least. but what if someone wanted the search results of just the works in that year? (ex: 1980c.e)

**Comment by jffcamp** [comment]
Count is not always right. Is that part of our estimate issue?
https://lux-front-dev.collections.yale.edu/view/concept/cc768f79-990b-46c7-83af-a99e1b0b6e08
Timeline -3600 to 1
1303 BCE - Objects Produced: 2
Object results, [when clicked, is 6](https://lux-front-dev.collections.yale.edu/view/results/objects?q={%22AND%22:[{%22OR%22:[{%22classification%22:{%22id%22:%22https://lux.collections.yale.edu/data/concept/cc768f79-990b-46c7-83af-a99e1b0b6e08%22}},{%22material%22:{%22id%22:%22https://lux.collections.yale.edu/data/concept/cc768f79-990b-46c7-83af-a99e1b0b6e08%22}}]},{%22producedDate%22:%22-1302%22,%22_comp%22:%22%3C%22},{%22producedDate%22:%22-1304%22,%22_comp%22:%22%3E%22}]}&collapseSearch=true)
Same date filter (-3600 to 1) - years with single result are impossible to click. 
Zoom in doesn't seem to do anything. Zoom out seems to clear the date range

**Comment by prowns** [comment]
Regarding Zoom feature - it's kind of confusing.

- change button labels to: Zoom In and See All
- only show one button (depending on state)

**Comment by kamerynB** [comment]
Moving this issue to blocked as it requires changes to support the pagination in facets.

**Comment by roamye** [comment]
@kamerynB - can this ticket be worked on for the 08/05 or 08/19  milestone as the [paginations issue](https://github.com/project-lux/lux-frontend/issues/252) was closed last week?

**Comment by kamerynB** [comment]
@roamye I've added it to the current milestone but I have to reach out to Kara to see if she has suggestions on color for the graph.

**Comment by roamye** [comment]
From IT meeting 08/02: 
Recharts comes with out of the box accessibility, but it ruins highlight to zoom. However, it does come with a brush component which allows someone to slide and see the date on the slider. This works with accessibility as they can use the arrow buttons on the keyboard to minimize maximize the slider from the brush component. 

This functionality can also allow the removal of date input fields (+2 Rob/Jeff).

**Comment by kamerynB** [comment]
Update 8/15:
I believe this just needs an accessibility review. I've let Kara know it is deployed but she is has more responsibilities now. @jffcamp Is there anyone else on the accessibility team that can provide a review of this?

**Comment by roamye** [comment]
DEV Mtg 10/14: 
There will need to be non LUX colors for the visualizations. Possibly patterns. (black/white/polka dots)

**Comment by kamerynB** [comment]
Reached out to Kara on 10/28. I came across [this resource](https://observablehq.com/@frankelavsky/contrast-and-no-use-of-color-alone-in-adjacent-charts) outlining possible accessibility solutions to charts.

**Comment by roamye** [comment]
@kamerynB - from your comment above, is this issue still blocked by an accessibility review?  Sounds like new solutions will need to be implemented before Kara can re-review the issue.

**Comment by prowns** [comment]
The key under the timeline looks like links.   Could/should we add a show all to let users see all objects produced by this person (for example)?  Or -- could that at least be noted as Timeline Key/Colors?

![image](https://github.com/user-attachments/assets/1969ab0f-b754-43b9-94d0-7fc91123091c)

**Comment by roamye** [comment]
Comments from @azaroth42 in teams thread ([here](https://teams.microsoft.com/l/message/19:351e0281aaa144d0812b960d216958c6@thread.tacv2/1732563532401?tenantId=dd8cbebb-2139-4df8-b411-4e3e87abeb5c&groupId=4620eac6-0955-4c69-93e8-b493b21f0df6&parentMessageId=1732563532401&teamName=ML%2F%20Cultural%20Heritage%20Support&channelName=1%20-%20LUX_ML&createdTime=1732563532401)) : 

- A border around the popup window to make it more visually distinct from the background, or a different background color?
- Styling on the links in the popup window to make it clearly you can click on them to get to the search results?
- If you haven't clicked, should the gray bar appear and popup appear? Currently if you slide the window, it appears by default on the left most entry?
- Should you be able to click somewhere else to get the popup to disappear, or should it have an [x] to close it?
- Colors could be brighter to be more visually distinct -- the green and blue look very similar, and the purple and black. I think contrasting is more important than not being garish.

**Comment by prowns** [comment]
@kamerynB  - can you turn off the vertical access labels?  https://lux-front-dev.collections.yale.edu/view/person/5e37df1d-e038-4cf7-a214-eb0422580013

**Comment by prowns** [comment]
@kamerynB - updated web accessible colors for timeline: https://www.figma.com/proto/bWS2orxFECNrG92HfG48BL/LUX-Yale-Design-Walkthrough?page-id=3833%3A482&node-id=3833-483&node-type=frame&viewport=660%2C471%2C0.17&t=JyhnryXAiyFGZMg9-1&scaling=min-zoom&content-scaling=fixed

**Comment by roamye** [comment]
Approved by UAT. 
Note: A separate ticket will be needed to capture new bugherds related to the visualizations.

**Comment by roamye** [comment]
Closing, looks good in PROD. New tickets will be made in upcoming ticket forming when bugherds are reviewed.
